### PR TITLE
(#9) New order by for version searching by version

### PR DESCRIPTION
**Pull request by gep13**
## Description Of Changes

This commit adds the Version property to SearchOrderBy, which can then be used within chocolatey/choco codebase, when required.

## Motivation and Context

When searching by version number in the 1.x version of Chocolatey, an additional orderby of "Id, Version desc" is included.  In order to maintain backwards compatibility, we need to also include this orderby in the new NuGet.Client based implementation.

This work is similar to the work that was done in this commit, where the DownloadCount orderby was added:

https://github.com/chocolatey/NuGet.Client/pull/25/commits/d28424ed2d2bf75afe77d775231c50ccc4a78c66

## Testing

1. Check out this PR, and build the NuGet.Client solution
2. Check out the associated PR into chocolatey/choco - https://github.com/chocolatey/choco/pull/3044
4. Pull the updated NuGet.Client assemblies into the chocolatey/choco project
5. Open Fiddler
6. Run the following command `search windirstat --version 1.1.2 --proxy http://localhost:8888`
7. Look at the captured requests in Fiddler, and ensure that the generated query includes the following `$orderby=Id,Version%20desc`

### Operating Systems Testing

- Windows 10/11

## Change Types Made

* [x] Bug fix (non-breaking change).
* [ ] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [ ] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [ ] Tests to cover my changes, have been added. (Tests has not been updated, as I do not know which are affected by localization and which may have been affected by the update, tests will instead be enabled or added to Chocolatey CLI E2E)
* [x] All new and existing tests passed? (More tests has been made English only since the last update, unable to verify whether they succeed or not)
* [ ] PowerShell code changes: PowerShell v2 compatibility checked?

## Related Issues

- #9  
- https://app.clickup.com/t/20540031/PROJ-511

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol/LegacyFeed/V2FeedQueryBuilder.cs [hunk 1]
```diff
@@ -306,6 +307,14 @@ private string BuildOrderBy(SearchOrderBy? searchOrderBy)
                     orderBy = string.Format("{0}%20desc,{1}", DownloadCountProperty, IdProperty);
                     break;
 
+                case SearchOrderBy.Version:
+                    orderBy = string.Format("{0},{1}%20desc", IdProperty, VersionProperty);
```
Comment (AdmiringWorm) [inline src/NuGet.Core/NuGet.Protocol/LegacyFeed/V2FeedQueryBuilder.cs:311]: I think I mentioned this during our call earlier, but to make sure.

Do we want to update the list resource to specify this order by default?

Comment (gep13) [reply src/NuGet.Core/NuGet.Protocol/LegacyFeed/V2FeedQueryBuilder.cs:311]: I was actually just looking at this... I have added a commit which I believe addresses this.

**Review by AdmiringWorm — APPROVED** [review]
LGTM

**Comment by AdmiringWorm** [comment]
@gep13 thanks for getting this added, great work 👍

**Pull request merged**